lib/oga/xml/parser.rll

```diff
@@ -245,7 +245,7 @@ string_body
   def reset
     @line = 1
 
-    @lexer.reset
+    @lexer.reset_native
   end
```

Comment:
That seems to have fixed it. I’ve pushed these changes and the tests all seem to pass now.